backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -239,6 +309,67 @@ def grant_target_role_access_policy(self):
             json.dumps(policy),
         )
 
+    def check_access_point_and_policy(self) -> bool:
+        """
+        Checks if access point created with correct permissions
+        :return: True if access point created with correct permissions else False
+        """
+        s3_client = S3ControlClient(self.source_account_id, self.source_environment.region)
+        access_point_arn = s3_client.get_bucket_access_point_arn(self.access_point_name)
+        if not access_point_arn:
+            self.folder_errors.append(format_error_message(None, None, None, "Access Point", self.access_point_name))
+            return
+
+        existing_policy = s3_client.get_access_point_policy(self.access_point_name)
+        if not existing_policy:
+            self.folder_errors.append(
+                format_error_message(None, None, None, "Access Point Policy", self.access_point_name)
+            )
+            return
+
+        existing_policy = json.loads(existing_policy)
+        statements = {item["Sid"]: item for item in existing_policy["Statement"]}
+        target_requester_id = SessionHelper.get_role_id(self.target_account_id, self.target_requester_IAMRoleName)
+        error = False
+        if f"{target_requester_id}0" not in statements.keys():
+            error = True
+        else:
+            prefix_list = statements[f"{target_requester_id}0"]["Condition"]["StringLike"]["s3:prefix"]
+            if isinstance(prefix_list, str):
+                prefix_list = [prefix_list]
+            if f"{self.s3_prefix}/*" not in prefix_list:
+                error = True
+        if error:
+            self.folder_errors.append(
+                format_error_message(
+                    self.target_requester_IAMRoleName,
+                    "Policy",
+                    f"{target_requester_id}0",
+                    "Access Point",
+                    self.access_point_name,
+                )
+            )
+
+        error1 = False
+        if f"{target_requester_id}1" not in statements.keys():
+            error1 = True
+        else:
+            resource_list = statements[f"{target_requester_id}1"]["Resource"]
+            if isinstance(resource_list, str):
+                resource_list = [resource_list]
+            if f"{access_point_arn}/object/{self.s3_prefix}/*" not in resource_list:
+                error1 = True
+        if error1:
+            self.folder_errors.append(
+                format_error_message(
+                    self.target_requester_IAMRoleName,
+                    "Policy",
+                    f"{target_requester_id}1",
+                    "Access Point",
+                    self.access_point_name,
+                )
+            )
+
```

Comment:
Updated the `def check_...` functions for each of the share managers to have a return type of `None` and updated docstring

No return statement needed for this functions since we are updating the arrays which contain a running list of errors found on the verify task